### PR TITLE
Expert Chapter 2

### PR DESCRIPTION
**Pull request by MuteTiefling**
### Misc

Helper for charging Certus earlier
image

Alternatives for a bunch of 1:1 magic crafts
image
image

image

Various charges crafted with their essence and tiny tnt
image

Fixed glass smelting (all modes) so that `#forge:sand/colorless` is used for regular glass and the other sand colors all smelt into colored glasses. They already did, but sand tag unification gave them all duplicate recipes, one for glass, and one for colored.

## Experimental
Added a pretty common chance to loot Stabilized Warp Scrolls from Blue Skies villages. They come pre-attuned to the Twilight Forest (0,200,0) as an emergency return to Twilight in case the player did not set up other means.

These are currently bugged, however. Working with Ars Nouveau and Twilight folks to figure out why. But the scrolls cannot teleport anyone into Twilight Forest. So if these don't work out, I'll need to come up with another method for travel between the dimensions. Ideally, something that a player can do if they lose all of their equipment and have their spawn locked to another dim... though I suppose they can also just break their bed and die again to yeet themselves back to Twilight 

## Powah
Removed Blazing and Hardened tier for remaining blocks. Now goes Basic > Niotic > Spirited > Nitro

Dramatically reduced the rates on all of these machines. Rates not final, just eyeballing them for now and will test more extensively later. 

In efforts to standardize on fewer machine frames, these will use Thermal frames
image
image

Energy Hopper
image

Upgrading requires the previous tier plus replacement parts:
image

Ender Gates / Cells will be available when wormholes are available for now. Still uncertain.
image
image

Personal Batteries
image

Energy Cells 
heavily considering removing energy storage entirely, but with batteries that can be charged/discharged it probably makes no sense to try
image

Energizing Rods and Upgrades
image
image

## Natures Aura
Finish removing crafts that use the disabled tokens
Eir's Token
image
Ring of Last Chance
image

Shifting Sundial
image
Hand of Time
image

Cloudshifter
image
Lamp of Sanctuary
image
Powder Manipulator
image

Altar of Birthing
Would like to do a pass on this and make it only passive animals at some point. 
image

The rest of these are probably temporary. I'm not fully sure where I want them in progression, but I also want to get rid of any remaining recipes using the tokens. 
Shooting Mark
image
Firecracker Gaze
image
Creational Catalyst
image

## Occultism
Bring Familiar Ring earlier
image
image

## Multi-Tools
With a heavier focus on manual mining, powerful tools need to be moved up.
First off, Thermal's tools are disabled in favor of IE's drill and saw. 

PNC Jackhammer
image

Various Drill bits
image
image
image

Final tier, which unlocks full vein mine, will probably come later. Recipe undecided. This should make the Jackhammer available towards the end of Chapter 1 with upgraded bits coming through early Chapter 2

IE Drill and Saw
These will be available shortly after Chapter 2 starts, as they're iron based. They'll also need liquid fuel. Considering limiting this to Abyssaline (See Tech progression below)
Drill
image

Saw
image

The tool heads are all unbreakable, so the only cost to running them is the fuel itself. 
image
image
image
image

Various upgrades
image
image
image
image

### CODE REVIEW

**Comment by MuteTiefling** [comment]
### Tech Progression

**Comment by MuteTiefling** [comment]
### Magic Progression

**Comment by MuteTiefling** [comment]
Think we're going to need to add Restricted Portals at this point. I just don't have a good way of controlling access out of the nether without it. 

https://www.curseforge.com/minecraft/mc-mods/restricted-portals

**Comment by MuteTiefling** [comment]
Gonna stop here as this cointains some changes to normal as well that should get into place. This effectively covers progression through Chapter 2 and a good chunk of tech balancing. Lots more side recipes still need adjusting.

File: kubejs/client_scripts/base/jei_descriptions.js [hunk 1]
```diff
@@ -27,25 +27,33 @@ JEIEvents.information((event) => {
...
+                `Bottled Sunlight: The Overworld, Twilight Forest, or the Everdawn`,
+                `Bottled Ghosts: Nether or Everbright`,
+                `Bottled Darkness: The End or The Bumblezone`,
                 ' ',
                 `Can be automated using a Dispenser.`
             ]
         },
         {
-            items: [Item.of('naturesaura:aura_bottle', { stored_type: 'naturesaura:end' })],
+            items: ['naturesaura:bottle_two_the_rebottling'],
```
Comment (NielsPilgaard) [inline kubejs/client_scripts/base/jei_descriptions.js:42]: 🤣

File: kubejs/server_scripts/base/recipes/lychee/block_interacting.js [hunk 2]
```diff
@@ -279,83 +279,6 @@ ServerEvents.recipes((event) => {
         }
     ];
 
-    const ae2_conversions = [
```
Comment (NielsPilgaard) [inline kubejs/server_scripts/base/recipes/lychee/block_interacting.js:282]: Was this added to AE2? :D

File: kubejs/server_scripts/expert/recipes/ars_nouveau/enchanting_apparatus.js [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    if (global.isExpertMode == false) {
+        return;
+    }
+
+    const id_prefix = 'enigmatica:expert/ars_nouveau/enchanting_apparatus/';
+    const recipes = [
+        {
+            output: { item: 'create:mechanical_crafter', count: 8 },
+            pedestalItems: [
+                { item: { item: 'create:brass_casing' } },
```
Comment (NielsPilgaard) [inline kubejs/server_scripts/expert/recipes/ars_nouveau/enchanting_apparatus.js:11]: `{ item: { item: 'create:brass_casing' } }` - is that the correct format?

**Review by NielsPilgaard — APPROVED** [review]

File: kubejs/server_scripts/base/recipes/lychee/block_interacting.js (hunk 2 above)
Comment (MuteTiefling) [reply kubejs/server_scripts/base/recipes/lychee/block_interacting.js:282]: Don't know if they've added it yet. I know they're talking about it though.

File: kubejs/client_scripts/base/jei_descriptions.js (hunk 1 above)
Comment (MuteTiefling) [reply kubejs/client_scripts/base/jei_descriptions.js:42]: Yeah, not really sure what happened here. The nbt types were working to show descriptions in JEI before (I think). This isn't ideal, but it's better than nothing.

File: kubejs/server_scripts/expert/recipes/ars_nouveau/enchanting_apparatus.js (hunk 3 above)
Comment (MuteTiefling) [reply kubejs/server_scripts/expert/recipes/ars_nouveau/enchanting_apparatus.js:11]: Yeah :-\

**Pull request merged**